File: android/app/src/main/java/jagtheora/vorbis/VorbisComment.java
package jagtheora.vorbis;

import jagtheora.misc.SimplePeer;

public final class VorbisComment extends SimplePeer {

    public VorbisComment() {
        this.init();
        if (this.b()) {
            throw new IllegalStateException();
        }
    }

    protected void init() {}

    protected void clear() {}
}

File: src/jvmMain/kotlin/jagtheora/vorbis/VorbisComment.kt
package jagtheora.vorbis

import jagtheora.misc.SimplePeer

actual class VorbisComment : SimplePeer() {
    init {
        init()
        check(!this.b())
    }

    actual external override fun clear()

    actual companion object {
        /*protected*/ actual external fun init()
    }
}
